### PR TITLE
Hotfix 7.1.10

### PR DESCRIPTION
**Pull request by ramonsmits**
Refactored in memory gateway deduplication implementation to be LRU based where LRU cache size can be configured by a code API.

### CODE REVIEW

File: src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplication.cs [hunk 1]
```diff
@@ -2,63 +2,47 @@
...
+                }
             }
         }
 
-        class GatewayMessage
-        {
-            public string Id { get; set; }
-            public DateTime TimeReceived { get; set; }
-        }
+        readonly int MaxSize;
```
Comment (bording) [inline src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplication.cs]: nit: Should be `maxSize`

File: src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayPersistence.cs [hunk 2]
```diff
@@ -8,14 +10,22 @@ public class InMemoryGatewayPersistence : Feature
...
         /// See <see cref="Feature.Setup" />.
         /// </summary>
         protected internal override void Setup(FeatureConfigurationContext context)
         {
-            context.Container.ConfigureComponent<InMemoryGatewayDeduplication>(DependencyLifecycle.SingleInstance);
+            var maxSize = context.Settings.Get<int>(MaxSizeKey);
+            context.Container.RegisterSingleton<IDeduplicateMessages>(new InMemoryGatewayDeduplication(maxSize));
         }
+
+        internal static string MaxSizeKey = "InMemoryGatewayDeduplication.MaxSize";
```
Comment (bording) [inline src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayPersistence.cs]: Why is this `internal static`? I believe this should be `public const`.

File: src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplication.cs [hunk 3]
```diff
@@ -2,63 +2,47 @@
...
-                return Task.FromResult(persistence.Add(new GatewayMessage
+                else
                 {
-                    Id = clientId,
-                    TimeReceived = timeReceived
-                }));
-            }
-        }
-
-        public int DeleteDeliveredMessages(DateTime until)
```
Comment (bording) [inline src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplication.cs]: So was this actually ever called anywhere? Seems strange that you could just delete this without needing to modify any calling code.

Comment (ramonsmits) [reply src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplication.cs]: That is correct @bording :-) dead code

**Pull request merged**